runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -2382,6 +2458,7 @@ public List<Windmill.KeyedGetDataRequest> getKeysToRefresh(Instant refreshDeadli
                       .setKey(shardedKey.key())
                       .setShardingKey(shardedKey.shardingKey())
                       .setWorkToken(work.getWorkItem().getWorkToken())
+                      .addAllLatencyAttribution(work.getLatencyAttributionList())
```

Comment:
is there a reason this is on KeyedGetDataRequest and not on work-item commit?

Reply:
StreamingDataflowWorker refreshes active work items every few seconds by sending empty KeyedGetDataRequests. We want to detect work items that are stuck on user processing, so we need regular updates WHILE processing. Sending ONLY an update on the commit isn't enough, but we've filed it as future work.